metrics/src/datapoint.rs

```diff
@@ -12,6 +12,12 @@
 //! The matric macro consists of the following three main parts:
 //!  - name: the name of the metric.
 //!
+//!  - tags (optional): when a metric sample is reported with tags, you can use
+//!    group-by when querying the reported samples.  Each metric sample can be
+//!    attached with zero to many tags.  Each tag is of the format:
+//!
+//!    - ("tag-name", "tag-value")
```

Comment:
I like the idea. My only bikesheding concern is that the tag syntax feels too close to the field syntax, and there'll likely be confusion.

Some alternative ideas:
* `"tag-name" => "tag-value"`
* `"tag-name" = "tag-value"`

Notably, the quotes are dropped so tags don't look similar to `("field-name", "field-value", ...)`

Reply:
Thank you.  I didn't know macro supports something like `=>`.  Will give it a try!

Reply:
It works!  Updated the comment and PR description as well.